vault/resource_pki_secret_backend_issuer.go

```diff
@@ -175,9 +175,16 @@ func pkiSecretBackendIssuerUpdate(ctx context.Context, d *schema.ResourceData, m
 
 	// only write to Vault if a patch is required
 	if patchRequired {
-		_, err := client.Logical().WriteWithContext(ctx, path, data)
-		if err != nil {
-			return diag.Errorf("error writing data to %q, err=%s", path, err)
+		if d.IsNewResource() {
+			_, err := client.Logical().WriteWithContext(ctx, path, data)
+			if err != nil {
+				return diag.Errorf("error writing issuer data to %q, err=%s", path, err)
+			}
+		} else {
+			_, err := client.Logical().JSONMergePatch(ctx, path, data)
+			if err != nil {
+				return diag.Errorf("error updating issuer data at %q, err=%s", path, err)
```

Comment:
Would be great if we could some how extend the test coverage to include these sorts of error conditions.

Reply:
Good call, added an extra test step to validate the error conditions. Thanks!